### PR TITLE
Solution of decorator

### PR DESCRIPTION
**Pull request by svyshcho**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    dict_of_args = {}
+
+    def inner(*args) -> Callable:
```
Comment (Bohdan-Salamakha) [inline app/main.py:7]: add typehint for args

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    dict_of_args = {}
+
+    def inner(*args: int) -> Callable:
```
Comment (Bohdan-Salamakha) [inline app/main.py:7]: *args is always a tuple cause * includes any count of objects in a tuple

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v2

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    dict_of_args = {}
+
+    def inner(*args: tuple) -> Callable:
+        if args in dict_of_args.keys():
```
Comment (olena-hul) [inline app/main.py:8]:
```suggestion
        if args in dict_of_args:
```

It will result in same as 

```if args in dict_of_args.keys()```

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**